File: frontend/src/types/memory.ts
export interface ComposeSessionInfo {
  id: string
  draft_preview_id: string
  session_type: string
  title: string
  editor_snapshot: Record<string, unknown>
  summary: string
  created_at: string
  updated_at: string
  messages: ComposeMessageInfo[]
}

export interface CreateComposeSessionRequest {
  draft_preview_id?: string
  session_type: string
  title: string
  editor_snapshot: Record<string, unknown>
}

export interface UpdateComposeSessionRequest {
  title?: string | null
  editor_snapshot?: Record<string, unknown> | null
}

export interface ComposeMessageRequest {
  role: 'user' | 'assistant' | 'system'
  content: string
  message_type?: 'normal' | 'summary'
  editor_snapshot?: Record<string, unknown>
}

export interface ComposeMessageInfo {
  id: string
  role: 'user' | 'assistant' | 'system'
  message_type: 'normal' | 'summary'
  content: string
  editor_snapshot: Record<string, unknown>
  archived: boolean
  created_at: string
}

File: backend/app/schemas/memory.py
"""第十三阶段写作记忆 API 结构。"""

from pydantic import BaseModel


class ComposeSessionInfo(BaseModel):
    id: str
    draft_preview_id: str = ""
    session_type: str
    title: str = ""
    editor_snapshot: dict = {}
    summary: str = ""
    created_at: str
    updated_at: str
    messages: list[dict] = []


class CreateComposeSessionRequest(BaseModel):
    draft_preview_id: str = ""
    session_type: str = "compose"
    title: str = ""
    editor_snapshot: dict = {}


class UpdateComposeSessionRequest(BaseModel):
    title: str | None = None
    editor_snapshot: dict | None = None


class ComposeMessageRequest(BaseModel):
    role: str
    content: str
    message_type: str = "normal"
    editor_snapshot: dict = {}


class ComposeMessageInfo(BaseModel):
    id: str
    role: str
    message_type: str = "normal"
    content: str
    editor_snapshot: dict = {}
    archived: bool = False
    created_at: str
